lib/bambora/v1/batch_payment_resource.rb

```diff
@@ -2,14 +2,48 @@
 
 module Bambora
   module V1
+    # Bank Electronic Funds Transfer (CAD) and Automatic Clearing House (USD)
     class BatchPaymentResource
-      # Summary: Bank Electronic Funds Transfer (CAD) and Automatic Clearing House (USD)
-      # Docs: https://dev.na.bambora.com/docs/guides/batch_payment/
-      #       https://dev.na.bambora.com/docs/references/batch_payment/
-      # Endpoint: https://api.na.bambora.com/v1/batchpayments
-      def initialize(client:, sub_path:)
+      attr_reader :client, :api_key, :sub_path
+
+      def initialize(client:, api_key:)
         @client = client
-        @sub_path = sub_path
+        @api_key = api_key
+        @sub_path = '/v1/batchpayments'
+      end
+
+      ##
+      # Post batch payment data.
+      #
+      # @example
+      #   batch_payment_resource.create(
+      #     [{
+      #       super_type: 'E',
+      #       transaction_type: 'D',
+      #       institution_number: 12345,
+      #       transit_number: 123,
+      #       account_number: 1223456789,
+      #       amount: 10000,
+      #       reference_nubmer: 1234,
+      #       reccipient_name: 'Hup Podling',
+      #       customer_code: '02355E2e58Bf488EAB4EaFAD7083dB6A',
+      #       dynamic_description: 'The Skeksis',
+      #     }],
+      #   )
+      #
+      # @param transactions [Array] of hashes with payment data.
+      # @param opts[:process_now] [Integer] optional. Defaults to 1.
+      # @param opts[:process_date] [String] optional. Must also set process_now to 0.
+      #
+      # @see https://dev.na.bambora.com/docs/references/batch_payment/
+      # @see https://dev.na.bambora.com/docs/guides/batch_payment/
+      def create(transactions, opts = { process_now: 1 })
```

Comment:
If the user includes their own `opts`, and `process_now` isn't included, should it be defaulted to `process_now: 1`? Or should we assume they have intentionally left it out?

Reply:
I should clarify ln 36. My understanding is that `process_now` overrides `process_date` but if `process_now` is absent and `process_date` is present, `process_now` is assumed to be `false`.

So, yes, we can assume they intentionally left it out. I'll update ln 36 to read `Must exclude process_now or pass { process_now: 0 }`.